pkg/util/admissioncheck/admissioncheck.go

```diff
@@ -0,0 +1,151 @@
+/*
+Copyright 2023 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package admissioncheck
+
+import (
+	"context"
+	"errors"
+	"fmt"
+
+	"k8s.io/apimachinery/pkg/runtime"
+	"k8s.io/apimachinery/pkg/runtime/schema"
+	"k8s.io/apimachinery/pkg/types"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/client/apiutil"
+
+	kueue "sigs.k8s.io/kueue/apis/kueue/v1beta1"
+)
+
+var (
+	ErrNilParametersRef = errors.New("nil parameters reference")
+	ErrBadParametersRef = errors.New("bad parameters reference")
+)
+
+type ConfigHelper[PCT interface {
+	client.Object
+	*CT
+}, CT any] struct {
+	gk     schema.GroupKind
+	client client.Client
+}
+
+func NewConfigHelper[PCT interface {
```

Comment:
What does `PCT` stand for? ProvisioningConfigType? I'm wondering if the `Provisioning` is obvious here, since the implementation in the file generally does not use it. Maybe the type should be called `ProvisioningConfigHelper` in that case?